src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -265,34 +278,37 @@ public void testExceptionOccurred(Assertion assertion, ExceptionResult exception
     public void close() {
     }
 
-    private UUID getAndOrStartSuite(WikiTestPage testPage) {
+
+    private void getAndOrStartSuite(WikiTestPage testPage) {
         String fullSuiteName = TestPageHelper.getFullSuiteName(testPage);
         String[] suites = fullSuiteName.split("\\.");
-        String suitePath = "";
+        List<String> suitesToCreate = new ArrayList<>();
+        List<String> suitePath =new ArrayList<>();
+        UUID parentSuiteId = null;
         UUID suiteId = null;
 
-        for (String suite : suites) {
-            UUID parentSuiteId = runContext.getSuiteId(suitePath);
-            suitePath = format("%s.%s", suitePath, suite);
-            suiteId = runContext.getSuiteId(suitePath);
-            if (suiteId == null) {
-                String description = null;
-                Set<Attribute> suiteAttrs = null;
-                PageData suitePageData = getPageDataForSuite(suitePath, testPage.getSourcePage());
-                if (suitePageData != null && suitePageData.getAttribute(WikiPageProperty.SUITES) != null) {
-                    suiteAttrs = convertAttributes(suitePageData.getAttribute(WikiPageProperty.SUITES));
-                }
-                if (suitePageData != null && suitePageData.getAttribute(WikiPageProperty.HELP) != null) {
-                    description = suitePageData.getAttribute(WikiPageProperty.HELP);
-                }
-
-                StartTestItem suiteItem = new StartTestItem(runContext.getTestRunUUID(), suite, TestItemType.SUITE, description, suiteAttrs);
-                suiteId = orangebeardClient.startTestItem(parentSuiteId, suiteItem);
-                runContext.addSuite(suitePath, suiteId, suiteItem.getStartTime());
+        for(String suite : suites) {
+            suitePath.add(suite);
+            suiteId = runContext.getSuiteId(String.join(".",suitePath));
+            if(suiteId != null) {
+                parentSuiteId=suiteId;
+            }else {
+                suitesToCreate.add(suite);
             }
         }
 
-        return suiteId;
+        if (!suitesToCreate.isEmpty()) {
+            StartSuiteRQ suiteItem = new StartSuiteRQ(runContext.getTestRunUUID(), parentSuiteId, new String("this is description"), new HashSet<Attribute>(), suitesToCreate); //description made min=1 in listenerapi
+            List<Suite> suiteList = v3client.startSuite(suiteItem);
+            for (Suite suite : suiteList) {
```

Comment:
I think you should adapt the runContext to the new suiteresponse, instead of jumping through hoops here to make the new response fit :thinking: